### PR TITLE
action #2996 - ensure well formed json for needles

### PR DESCRIPTION
**Pull request by psladek**
explicit hash checking via decode_json() import

### CODE REVIEW

**Comment by coolo** [comment]
test suite is broken

File: lib/OpenQA/Controller/Step.pm [hunk 1]
```diff
@@ -391,6 +391,58 @@ sub _timestamp {
...
+            $errorh{'message'}="area without type";
+            return \%errorh;
+        }
+        if (!exists $area->{'height'} ) {
+            $errorh{'message'}="area without height";
+            return \%errorh;
+        }
+        if (!exists $area->{'width'} ) {
+            $errorh{'message'}="area without width";
+            return \%errorh;
```
Comment (aaannz) [inline lib/OpenQA/Controller/Step.pm]: Why are you returing hashref if you only use that for sotring the string? Why not return the error message itself?

Comment (psladek) [inline lib/OpenQA/Controller/Step.pm]: Idea was to return a richer structure, in the end the message was enough. Not really a problem as it is now, and allows easy extension if needed.

Comment (aaannz) [inline lib/OpenQA/Controller/Step.pm]: Well it looks rather criptic and thus it is kind of problem for me. Can be easily changed to hashref if needed in the future. ;)

Comment (coolo) [inline lib/OpenQA/Controller/Step.pm]: agreed

Comment (psladek) [inline lib/OpenQA/Controller/Step.pm]: well, if this is cryptic, I do not know how to call the rest of most of the openQA code... but ok, no problem to change it.

**Comment by coolo** [comment]
LGTM

**Pull request merged**